File: php/features.php
<?php
declare(strict_types=1);

// Virenmonitoring SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class VirenmonitoringFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new VirenmonitoringBaseFeature();
            case "test":
                return new VirenmonitoringTestFeature();
            default:
                return new VirenmonitoringBaseFeature();
        }
    }
}

File: py/features.py
# Virenmonitoring SDK feature factory

from feature.base_feature import VirenmonitoringBaseFeature
from feature.test_feature import VirenmonitoringTestFeature


def _make_feature(name):
    features = {
        "base": lambda: VirenmonitoringBaseFeature(),
        "test": lambda: VirenmonitoringTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
